libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpVP9Reader.java

```diff
@@ -0,0 +1,230 @@
+        isKeyFrame = (data.peekUnsignedByte() & 0x04) == 0;
+      }
+
+      if (!isOutputFormatSet && width > 0 && height > 0) {
+        if (width != payloadFormat.format.width || height != payloadFormat.format.height) {
+          trackOutput.format(
+              payloadFormat.format.buildUpon().setWidth(width).setHeight(height).build());
+        }
+        isOutputFormatSet = true;
+      }
+
+      int fragmentSize = data.bytesLeft();
+      // Write the video sample.
+      trackOutput.sampleData(data, fragmentSize);
+      fragmentedSampleSizeBytes += fragmentSize;
+
+      if (rtpMarker) {
+        if (firstReceivedTimestamp == C.TIME_UNSET) {
+          firstReceivedTimestamp = timestamp;
+        }
+        bufferFlags = isKeyFrame ? C.BUFFER_FLAG_KEY_FRAME : 0;
+        long timeUs = toSampleUs(startTimeOffsetUs, timestamp, firstReceivedTimestamp);
+        trackOutput.sampleMetadata(
+            timeUs,
+            bufferFlags,
+            fragmentedSampleSizeBytes,
+            /* offset= */ 0,
+            /* encryptionData= */ null);
+        fragmentedSampleSizeBytes = 0;
+        gotFirstPacketOfVP9Frame = false;
+      }
+      previousSequenceNumber = sequenceNumber;
+    }
+  }
+
+  @Override
+  public void seek(long nextRtpTimestamp, long timeUs) {
+    firstReceivedTimestamp = nextRtpTimestamp;
+    fragmentedSampleSizeBytes = 0;
+    startTimeOffsetUs = timeUs;
+  }
+
+  // Internal methods.
+  private static long toSampleUs(
+      long startTimeOffsetUs, long rtpTimestamp, long firstReceivedRtpTimestamp) {
+    return startTimeOffsetUs
+        + Util.scaleLargeTimestamp(
+            (rtpTimestamp - firstReceivedRtpTimestamp),
+            /* multiplier= */ C.MICROS_PER_SECOND,
+            /* divisor= */ MEDIA_CLOCK_FREQUENCY);
+  }
+
+  private static boolean parseVP9Descriptor(ParsableByteArray payload, int packetSequenceNumber)
+      throws ParserException {
+    // VP9 Payload Descriptor, Section 4.2
+    //         0 1 2 3 4 5 6 7
+    //        +-+-+-+-+-+-+-+-+
+    //        |I|P|L|F|B|E|V|Z| (REQUIRED)
+    //        +-+-+-+-+-+-+-+-+
+    //   I:   |M| PICTURE ID  | (RECOMMENDED)
+    //        +-+-+-+-+-+-+-+-+
+    //   M:   | EXTENDED PID  | (RECOMMENDED)
+    //        +-+-+-+-+-+-+-+-+
+    //   L:   | TID |U| SID |D| (Conditionally RECOMMENDED)
+    //        +-+-+-+-+-+-+-+-+
+    //        |   TL0PICIDX   | (Conditionally REQUIRED)
+    //        +-+-+-+-+-+-+-+-+
+    //   V:   | SS            |
+    //        | ..            |
+    //        +-+-+-+-+-+-+-+-+
+
+    int header = payload.readUnsignedByte();
+    if (!gotFirstPacketOfVP9Frame) {
+      // For start of VP9 partition B=1 as per VP9 RFC Section 4.2.
+      if ((header & 0x08) == 0) {
+        Log.w(
+            TAG,
+            Util.formatInvariant(
+                "First payload octet of the RTP packet is not the beginning of a new VP9 partition,"
+                    + " Dropping current packet." + header));
```

Comment:
I don't find it necessary to log the header here.

- If you think it's necessary, please log with format string, like `Util.formatInvariant("xxxx %d", header);` instead of using `+`.
- else, just log the string and remove `Util.formatInvariant`